File: front-end/src/service/api/product.ts
import { http } from '../http';

// 产品列表查询参数
export interface ProductQueryParams {
  page?: number;
  pageSize?: number;
  categoryID?: string;
  name?: string;

}

// 产品创建请求
export interface ProductCreateRequest {
  categoryID?: string;
  name: string;
  sort?: number;

}

// 产品更新请求
export interface ProductUpdateRequest {
  id: number;
  categoryID?: string;
  name?: string;
  sort?: number;

}

// 产品响应
export interface ProductResponse {
  id: number;
  createdAt: string;
  updatedAt: string;
  categoryID: string;
  name: string;
  sort: number;

}

// 产品列表响应
export interface ProductListResponse {
  total: number;
  list: ProductResponse[];
}

// 获取产品列表
export const getProducts = (params: ProductQueryParams) => {
  return http.request<ProductListResponse>({
    url: '/product/list',
    method: 'GET',
    params,
  });
};

// 获取产品详情
export const getProduct = (id: number) => {
  return http.request<ProductResponse>({
    url: '/product/detail/' + id,
    method: 'GET',
  });
};

// 创建产品
export const createProduct = (data: ProductCreateRequest) => {
  return http.request<void>({
    url: '/product/create',
    method: 'POST',
    data,
  });
};

// 更新产品
export const updateProduct = (data: ProductUpdateRequest) => {
  return http.request<void>({
    url: '/product/update',
    method: 'PUT',
    data,
  });
};

// 删除产品
export const deleteProduct = (id: number) => {
  return http.request<void>({
    url: '/product/delete/' + id,
    method: 'DELETE',
  });
};

File: server/apps/admin/dto/product.go
package dto



// ProductCreateRequest 创建产品请求
type ProductCreateRequest struct {

	CategoryID string `json:"categoryID" ` // 产品分类

	Name string `json:"name" binding:"required"` // 名称

	Sort int64 `json:"sort" ` // 排序

}

// ProductUpdateRequest 更新产品请求
type ProductUpdateRequest struct {
	ID uint `json:"id" binding:"required"` // ID

	CategoryID string `json:"categoryID" ` // 产品分类

	Name string `json:"name" ` // 名称

	Sort int64 `json:"sort" ` // 排序

}

// ProductQueryParams 产品查询参数
type ProductQueryParams struct {
	Page     int `form:"page"`      // 页码
	PageSize int `form:"pageSize"`  // 每页条数


	CategoryID string `form:"categoryID"` // 产品分类

	Name string `form:"name"` // 名称

}

// ProductResponse 产品响应
type ProductResponse struct {
	ID        uint   `json:"id"`         // ID
	CreatedAt string `json:"createdAt"`  // 创建时间
	UpdatedAt string `json:"updatedAt"`  // 更新时间

	CategoryID string `json:"categoryID"` // 产品分类

	Name string `json:"name"` // 名称

	Sort int64 `json:"sort"` // 排序

}

// ProductListResponse 产品列表响应
type ProductListResponse struct {
	Total int64                  `json:"total"`  // 总数
	List  []*ProductResponse `json:"list"`    // 列表
}
